test/integration/client-side-operations-timeout/node_csot.test.ts

```diff
@@ -576,6 +581,201 @@ describe('CSOT driver tests', metadata, () => {
     });
   });
 
+  describe('Tailable cursors', function () {
+    let client: MongoClient;
+    let internalClient: MongoClient;
+    let commandStarted: CommandStartedEvent[];
+    const metadata: MongoDBMetadataUI = {
+      requires: { mongodb: '>=4.4' }
+    };
+
+    const failpoint: FailPoint = {
+      configureFailPoint: 'failCommand',
+      mode: 'alwaysOn',
+      data: {
+        failCommands: ['aggregate', 'find', 'getMore'],
+        blockConnection: true,
+        blockTimeMS: 100
+      }
+    };
+
+    beforeEach(async function () {
+      internalClient = this.configuration.newClient();
+      await internalClient
+        .db('db')
+        .dropCollection('coll')
+        .catch(() => null);
+
+      await internalClient.db('db').createCollection('coll', { capped: true, size: 1_000_000 });
+
+      await internalClient
+        .db('db')
+        .collection('coll')
+        .insertMany(
+          Array.from({ length: 100 }, () => {
+            return { x: 1 };
+          })
+        );
+
+      await internalClient.db().admin().command(failpoint);
+
+      client = this.configuration.newClient(undefined, { monitorCommands: true, minPoolSize });
+      commandStarted = [];
+      client.on('commandStarted', ev => commandStarted.push(ev));
+      await client.connect();
+    });
+
+    afterEach(async function () {
+      await internalClient
+        .db()
+        .admin()
+        .command({ ...failpoint, mode: 'off' });
+      await internalClient.close();
+      await client.close();
+    });
+
+    context('when in ITERATION mode', function () {
+      context('awaitData cursors', function () {
+        let cursor: FindCursor;
+        afterEach(async function () {
+          if (cursor) await cursor.close();
+        });
+
+        it('applies timeoutMS to initial command', metadata, async function () {
+          cursor = client
+            .db('db')
+            .collection('coll')
+            .find({}, { timeoutMS: 50, tailable: true, awaitData: true, batchSize: 1 });
+          const maybeError = await cursor.next().then(
+            () => null,
+            e => e
+          );
+          expect(maybeError).to.be.instanceOf(MongoOperationTimeoutError);
```

Comment:
nothing here asserts that the aggregate specifically timed out, only that cursor execution timed out.

Perhaps we could assert on the commandStartedEvents and confirm that there was only one command sent?